src/main/java/org/scijava/io/DataHandle.java

```diff
@@ -74,43 +82,86 @@
 	void setLength(long length) throws IOException;
 
 	/**
+	 * Verifies that the handle has sufficient bytes available to read, returning
+	 * the actual number of bytes which will be possible to read, which might
+	 * be less than the requested value.
+	 * 
+	 * @param count Number of bytes to read.
+	 * @return The actual number of bytes available to be read.
+	 * @throws IOException If something goes wrong with the check.
+	 */
+	default long available(final long count) throws IOException {
```

Comment:
I almost wrote here that we shouldn't name this method `available`. The reason being that for `InputStream`, `available` does _not_ equate to "number of bytes till EOF" but rather "number of bytes which could currently be read from the stream." However, I think the intent there is that you can have a stream which is producing bytes in real time—you may read up to the _current_ end of stream, but more might appear later. So actually, the question really is: __Do we want to support increasingly-long `Location` types? Or only statically sized `Location`s?__ The original use case definitely assumes only statically sized things—i.e., constant `length()`. I think it is safest and best to retain this assumption, and document it carefully. Do you agree @gab1one?